sdk/core/azure-core/src/main/java/com/azure/core/credentials/BasicAuthenticationCredential.java

```diff
@@ -44,10 +44,11 @@ public BasicAuthenticationCredential(String username, String password) {
         String credential = username + ":" + password;
         String encodedCredential;
         try {
+            // Please change the error message everytime you update the named charset.
             encodedCredential = Base64Util.encodeToString(credential.getBytes("UTF8"));
         } catch (UnsupportedEncodingException e) {
             // The encoding is hard-coded, so if it's unsupported, it needs to be fixed right here.
-            throw logger.logExceptionAsError(new RuntimeException(e));
+            throw logger.logExceptionAsError(new RuntimeException("Please fixed the unsupported encoding UTF8.", e));
```

Comment:
Usually, the message in exceptions should inform what went wrong and shouldn't include corrective actions to fix the error. In this case, there's nothing the user could do to fix it as the encoding format is hardcoded in this method. It's better to revert this change and let the message in `UnsupportedEncodingException` be used by the logger which is more informative.

Reply:
Good point. Will reword that